ui/kotlinx-coroutines-javafx/src/JavaFxConvert.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ *  Copyright 2016-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines.javafx
+
+import javafx.beans.value.ChangeListener
+import javafx.beans.value.ObservableValue
+import kotlinx.coroutines.*
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.*
+
+/**
+ * Creates an instance of a cold [Flow] that subscribes to the given [ObservableValue] and produces
+ * its values as they change.
+ *
+ * The resulting flow is conflated, meaning that if several values arrive in a quick succession, only
+ * the last one will be produced.
+ *
+ * It produces at least one value.
```

Comment:
>It produces at least one value.

This guarantee is a bit weird and leads to an inconvenient API.
Basically, it means that one **cannot** have `ObservableValue<ObjectSubtype>` without a default value (a common case in UI) and non-null bound.
Have you considered an option to ignore the very first value (line 42) if it is a `null` like Rx adapter does?

Reply:
>  Basically, it means that one cannot have `ObservableValue<ObjectSubtype>` without a default value (a common case in UI) and non-null bound.

True, but I don't see how this would be possible in any case:
https://docs.oracle.com/javase/8/javafx/api/javafx/beans/value/ObservableValue.html#getValue-- is it even allowed to implement an `ObservableValue` that throws an exception on calls to `getValue()`?

> This guarantee is a bit weird and leads to an inconvenient API.

The reason why I even added this guarantee was convenience: this would allow one to collect this flow in a thread aside from the JavaFX main thread and always have an initial value to work with, not worrying with synchronization. Otherwise, in order to always have an up-to-date version of an `ObservableValue` in a separate thread, one would have to block the JavaFX thread, which is a no-go.

>  Have you considered an option to ignore the very first value (line 42) if it is a null like Rx adapter does?

This would violate the guarantee that at least one value is produced, and if that guarantee does not hold, what would be a good reason to emit a value at all?

In short, if there is such a use case—collecting from a separate thread and being sure that you are not terribly out-of-date—then, I believe, at least one value should be emitted; otherwise, I don't see why emit an initial value in any case, but the solution with sending an initial value only if it is not null is seemingly not worse.